### ISSUE TITLE
"invoke local" doesn't do anything

### ISSUE THREAD

**Issue by kennedyrose** [serverless/serverless-google-cloudfunctions#129]
I'm using the [default Node.js template](https://serverless.com/framework/docs/providers/google/guide/quick-start/). Running invoke works perfectly fine on the deployed application, but running `sls invoke local -f first` doesn't do anything or print anything to the console.

When logging debug info no errors are shown. It gets to `Serverless: Invoke invoke:local` then the process just exits.

**Comment by claytonpiccinin** [comment]
Same here.

Any updates about it?

**Comment by evandrom** [comment]
Same here.

Any updates about it?

**Comment by fperron01** [comment]
Is there a workaround for this issue ?

**Comment by kindofbeard** [comment]
Just ran into this problem, it seems to happen when two conditions are met: 

- you run the `sls invoke local` command via a debugger (Vscode in my case)
- no input data was provided to `invoke local` (via either  `data` or `path` parameters). 

In that case, `InvokeLocal` plugin gets stuck trying to read data on stdin. The problem seems to be located in `get-stdin/index.js`, where the condition `if (stdin.isTTY)`" is false. May be related to the way VScode calls programs to debug?! 

An easy workaround is to add a `--data` or `--path` parameter to your `invoke local` command.

**Comment by evandrom** [comment]
this workaround worked for you?
I tried and fails here :(

**Comment by buffolander** [comment]
This one doesn't work for me as well

**Comment by kindofbeard** [comment]
Well yes I confirm, the following config "just works" for me right now: 

`    {
      "type": "node",
      "request": "launch",
      "name": "Test running lambda sls",
      "program": "${workspaceFolder}/node_modules/.bin/sls",
      "args": [
        "invoke",
        "local",
        "-f",
        "myLambdaFunction",
        "--data",
        "dummyData"
      ]
    } `

If I remove the "--data dummyData" section, Invoke Lambda gets stuck as before. 
Latest Vscode, and my OS is an Ubuntu 18.04 LTS in case it matters...

**Comment by johku90** [comment]
> Just ran into this problem, it seems to happen when two conditions are met:
> 
> * you run the `sls invoke local` command via a debugger (Vscode in my case), and:
> * no input data was provided to `invoke local` (via either  `data` or `path` parameters).
> 
> In that scenario, `InvokeLocal` plugin gets stuck trying to read data on stdin. The problem seems to be located in `get-stdin/index.js`, where the condition `if (stdin.isTTY)` is false for some reason. May be related to the way VScode calls programs to debug?!
> 
> An easy workaround is to add a dummy `--data` or `--path` parameter to your `invoke local` command.

Yeah, this workaround works for me also.  I'm using windows 10 and git bash

**Comment by sadowskipawel** [comment]
Any update for this?
Workaround dont work for me.
When i invoke in shell

`serverless invoke local -f first --data '{"name": "Bernie"}'`

nothing happens.
this doesn't do anything or print anything to the console

**Comment by parkan** [comment]
I am experiencing the same thing, except rather than hanging it just exits after "Serverless: Invoke invoke:local" with no output or even attempting to compile the code (introducing a deliberate syntax error doesn't have any effect). Seems like maybe this plugin is not ready for use?

**Comment by scarolan** [comment]
Same issue here. `serverless invoke local` fails to produce any output, whereas the deployed function does output "Hello world".  I am using Google Cloud Shell for my dev environment.

**Comment by prakis** [comment]
try this (assuming hello is your function name)

`sls invoke local -f hello -d`

**Comment by neoromantic** [comment]
Can't get it to work either.

**Comment by kkuchta** [comment]
I assume this isn't the problem any of you all are running into, but since I had the exact same symptoms (`sls invoke local -f user-endpoint --data "{}" -l` runs for a second and returns with no output), here was my issue:

I was running the command from the wrong director.  As soon as I `cd`'d to the the directory containing my serverless.yml file, it worked fine.  🤦‍♂️

**Comment by hammadzz** [comment]
Same issue, I am using it alongside "serverless-plugin-typescript", after a lot of attempts to fix it the local invocation does not work. It compiles but will not invoke.
